datalad/support/tests/test_external_versions.py

```diff
@@ -262,3 +263,18 @@ def test_check():
 
     with assert_raises(OutdatedExternalDependency):
         ev.check('datalad', min_version="10000000")  # we will never get there!
+
+
+def test_add():
+    ev = ExternalVersions()
+    ev.add('custom1', lambda: "0.1.0")
+    assert_in("custom1=0.1.0", ev.dumps(query=True))
+    assert_not_in("numpy", ev.INTERESTING)  # we do not have it by default yet
+    assert_not_in("numpy=", ev.dumps(query=True))
+    ev.add('numpy')
+    assert_in("numpy=", ev.dumps(query=True))
```

Comment:
Fails on appveyor: https://ci.appveyor.com/project/mih/datalad/builds/32511871/job/ddbp8p80apy7i34r#L1038

I can trigger the same failure locally.  I think the issue is that `__getitem__` has an early return of None on an import error:

https://github.com/datalad/datalad/blob/d9aaae807d279d95d93f2f022541ecb2cecabd35/datalad/support/external_versions.py#L220-L244

So `._version.keys()` will not have "numpy" present if numpy isn't installed in the environment.  I think this test was assuming that the output would include "numpy=UNKNOWN" in this case.  This feels like a logic bug in external_versions (exposed, not added, by this PR).

Reply:
THANK YOU for the analysis!  I first thought that indeed code is wrong but then decided that  code is correct as is: We do have a notion of "UNKNOWN" version but that is when the external **is** present but we do not know its version.  If external is not present, then `__getitem__` returns None, as to signal that it is really None, i.e. no any kind of version of the external is available.
I have pushed a fix to the test in d4dcc2c4db2b7c7b9ed9081fa65c0b0984cc12a4 . Hopefully this would make this what should have been a trivial PR finally ready ;-)